src/test/java/SearchServletTest.java

```diff
@@ -184,6 +184,29 @@ public void queryWithoutStoreAndCategory() throws IOException {
     Assert.assertTrue(stringWriter.toString().contains("\"id\":3"));
   }
 
+  @Test
+  public void queryAllReceipts() throws IOException {
+    // Columns ommitted from database visual: id, userId, blobKey, imageUrl, rawText.
+    //
+    // id   Timestamp      Price          Store                    Categories
+    // 1  1045237591000    26.12        "walmart"         ["candy", "drink", "personal"]
+    // 2  1560193140000    14.51        "contoso"         ["cappuccino", "sandwich", "lunch"]
+    // 3  1491582960000    29.01   "main st restaurant"   ["food", "meal", "lunch"]
+
+    // Add mock receipts to datastore.
+    TestUtils.addTestReceipts(datastore);
+
+    // Perform doGet - this should retrieve all receipts.
+    when(request.getParameter("isNewLoad")).thenReturn("true");
+    servlet.doGet(request, response);
+    writer.flush();
+
+    // Make sure all receipts retrieved by finding the ids in the writer.
```

Comment:
If someone added another receipt to `addTestReceipts()`, this test would still pass but it would not have checked that all receipts had been returned right? I think I might put all of the receipts (aka BlobKeys) in `addTestReceipts()` in a static list and this test can call `addTestReceipts(datastore, receipts)` with the receipts to add. Then the assert can check receipts match.

Reply:
BlobKeys?

Reply:
Sorry I meant `Entity`s (note none of this code is tested and you would need a small Receipt wrapper class):

```
private static final Entity RECEIPT_ENTRY_1 = createEntity(new BlobKey("test"), "img/walmart-receipt.jpg", 26.12, "walmart", ImmutableSet.of("candy", "drink", "personal"), "");
...

private static final ImmutableList<Entity> receiptEntries = ImmutableSet.of(RECEIPT_ENTRY_1, ....);
..
addTestReceipts(datastore, receiptEntries);
```
Then in the test:

```
Receipt[] receipts = new Gson().fromJson(stringWriter.toString(), Receipt[].class);
Assert.assertTrue(receipts.size(), receiptEntries.length());
Assert.assertTrue(receipts.stream().anyMatch(...)); // check id
...
...
```